Marlin/src/inc/Conditionals_post.h

```diff
@@ -558,7 +558,9 @@
   #endif
 
   #if HAS_SD_DETECT && NONE(HAS_GRAPHICAL_TFT, LCD_USE_DMA_FSMC, HAS_FSMC_GRAPHICAL_TFT, HAS_SPI_GRAPHICAL_TFT, IS_DWIN_MARLINUI, EXTENSIBLE_UI, HAS_DWIN_E3V2)
-    #define REINIT_NOISY_LCD 1  // Have the LCD re-init on SD insertion
+    #if U8G_PARAM != (U8G_I2C_OPT_NONE | U8G_I2C_OPT_FAST) // But not for I2C LCDs
```

Comment:
It's not guaranteed that `U8G_PARAM` will be defined at this point, but this can be moved to `marlinui.h` where `ui.reinit_lcd()` is defined.